src/PIL/TiffImagePlugin.py

```diff
@@ -1577,6 +1577,9 @@ def _save(im, fp, filename):
     # aim for 64 KB strips when using libtiff writer
     if libtiff:
         rows_per_strip = min((2 ** 16 + stride - 1) // stride, im.size[1])
+        # JPEG encoder expects multiple of 8 rows
+        if compression == "jpeg":
+            rows_per_strip = min(((rows_per_strip + 7) // 8) * 8, im.size[1])
```

Comment:
```suggestion
            rows_per_strip = ((rows_per_strip + 7) // 8) * 8
```
Because `rows_per_strip` has already been set to be `im.size[1]` at minimum, and the calculation here only ever rounds up, the `min` part is unnecessary.

Reply:
I believe it is needed, precisely because it rounds up. This is to support the rare case of small images where im.size[1]<8 and you end up as a single strip. Libtiff/jpeg codec have no problem handling the single/last strip that is not factor of 8, as it is padded internally.

Reply:
Ok, thanks

Reply:
```suggestion
            rows_per_strip = min(((rows_per_strip + 7) // 8) * 8, rows_per_strip)
```
I think this is clearer?

Reply:
That won't work for the corner case where you have u multi-strip w/ e.g. rows_per_strip=2 and im.size[1]=6. For JPEG, you want to end up as single strip of 6 rows.